### PR TITLE
TST: Sparse matrix test with int64 indptr and indices

### PR DESCRIPTION
**Pull request by sethtroisi**
I don't like commented code; either we should remove the comment or make it work.

This was my attempt to test what I believe the commented test was supposed to test.

### CODE REVIEW

File: scipy/sparse/tests/test_base.py [hunk 1]
```diff
@@ -3544,6 +3527,35 @@ def test_constructor6(self):
         assert_array_equal(csr.shape, (3,6))
         assert_(np.issubdtype(csr.dtype, np.signedinteger))
 
+    def test_constructor_smallcol(self):
+        # int64 indices not required
+        data = arange(6) + 1
+        col = array([1, 2, 1, 0, 0, 2], dtype='int64')
+        ptr = array([0, 2, 4, 6], dtype='int64')
```
Comment (perimosocordiae) [inline scipy/sparse/tests/test_base.py:3534]: Use `np.int64` instead of the string version? (Here and elsewhere)

**Review by perimosocordiae — APPROVED** [review]
Looks fine to me. Can we find the commit that commented out this test and check if there's something we're missing here?

File: scipy/sparse/tests/test_base.py (hunk 1 above)
Comment (sethtroisi) [reply scipy/sparse/tests/test_base.py]: Done.

**Comment by sethtroisi** [comment]
It was added ~13 years ago in https://github.com/scipy/scipy/commit/d6b57de7c639e57532dfde1e826c505df823c366 which tried to unify constructors

**Review by rgommers — APPROVED** [review]
LGTM, and the commit from 13 years ago simply disabled 64-bit indices, which should work fine now. Let's get this in. Thanks @sethtroisi and @perimosocordiae

**Pull request merged**